src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/_params.py

```diff
@@ -48,6 +50,9 @@ def load_arguments(self, _):
         c.argument('proj_name', help='Name of the start up project file name. Required only for C#.')
         c.argument('sln_name', help='Name of the start up solution file name. Required only for C#.')
         c.argument('code_dir', options_list=['--code-dir'], help='The directory to download deployment scripts to.')
+        c.argument('version', options_list=['-v', '--version'], help='The Microsoft Bot Builder SDK version to be used '
+                                                                     'to create the bot',
+                   arg_type=get_enum_type(['v3', 'v4']), arg_group='Web/Function bot Specific')
```

Comment:
What is the purpose of this? The stands out as unusual.

Reply:
There is a difference in the prepare-publish steps for v3 bots versus v4 bots, namely in the exclusion of the deploy.cmd.template file for v4 bots. The optional argument `--version` allows for a quick break before performing any network calls and in this case, informs the user of the correct command to call.

If the user doesn't specify the version, we default to v3 but place a try-except around the code that will blow up if the bot is actually a v4 bot.

Reply:
People use this to publish bots, and when the bot is of kind webapp, we also publish a website to host their bot. We are transitioning from Bot Framework V3 to V4 and there are large user bases in both versions now, so we allow users to choose whether we'll publish a v3 or v4 bot template to their website.

Reply:
Perhaps update the text then to read that it is the version of the bot template to deploy.

Reply:
Sure, that makes sense.

Reply:
I pushed this change btw.